Source/JavaScriptCore/runtime/SamplingProfiler.cpp

```diff
@@ -1085,11 +1092,19 @@ Ref<JSON::Value> SamplingProfiler::stackTracesAsJSON()
         processUnverifiedStackTraces();
     }
 
+    HashMap<SourceID, Ref<SourceProvider>> sources;
+
     auto stackFrameAsJSON = [&](StackFrame& stackFrame) {
+        auto sourceID = stackFrame.sourceID();
+        if (RefPtr sourceProvider = stackFrame.sourceProvider())
```

Comment:
nit: This queries sourceProvider twice.  Would it be beneficial to just have a `sourceProviderAndID` method that returns a tuple instead?

Reply:
Changed.